blockproducer/metastate.go

```diff
@@ -1017,29 +1035,68 @@ func (s *metaState) transferSQLChainTokenBalance(transfer *types.Transfer) (err
 	realSender, err := crypto.PubKeyHash(transfer.Signee)
 	if err != nil {
 		err = errors.Wrap(err, "applyTx failed")
-		return err
+		return
 	}
 
 	if realSender != transfer.Sender {
 		err = errors.Wrapf(ErrInvalidSender,
 			"applyTx failed: real sender %s, sender %s", realSender.String(), transfer.Sender.String())
 		log.WithError(err).Warning("public key not match sender in applyTransaction")
+		return
 	}
 
 	var (
 		sqlchain *types.SQLChainProfile
 		ok       bool
 	)
-	sqlchain, ok = s.loadSQLChainObject(transfer.Sender.DatabaseID())
+	sqlchain, ok = s.loadSQLChainObject(transfer.Receiver.DatabaseID())
 	if !ok {
-		return ErrDatabaseNotFound
+		err = ErrDatabaseNotFound
+		log.WithFields(log.Fields{
+			"dbid":   transfer.Receiver.DatabaseID(),
+			"sender": transfer.Sender.String(),
+		}).WithError(err).Warning("database not exist in transferSQLChainTokenBalance")
+		return
+	}
+	if sqlchain.TokenType != transfer.TokenType {
+		err = ErrWrongTokenType
+		log.WithFields(log.Fields{
+			"dbid":   transfer.Receiver.DatabaseID(),
+			"sender": transfer.Sender.String(),
+		})
+		return
 	}
 
 	for _, user := range sqlchain.Users {
 		if user.Address == transfer.Sender {
-			if sqlchain.TokenType != transfer.TokenType {
-				return ErrWrongTokenType
+			// process arrears
+			if user.Arrears > 0 {
+				if user.Arrears <= transfer.Amount {
+					for _, miner := range sqlchain.Miners {
+						newUserArrears := make([]*types.UserArrears, len(miner.UserArrears))
+						i := 0
+						for _, ua := range miner.UserArrears {
+							if ua.User == user.Address {
+								miner.PendingIncome += miner.UserArrears[i].Arrears
+							} else {
+								newUserArrears[i] = ua
+								i += 1
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
should replace i += 1 with i++